platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/transition/execution/auto/WorkflowAutoTransitionCascadedExecutor.java

```diff
@@ -0,0 +1,69 @@
+package com.softicar.platform.workflow.module.workflow.transition.execution.auto;
+
+import com.softicar.platform.common.core.exceptions.SofticarException;
+import com.softicar.platform.workflow.module.workflow.item.AGWorkflowItem;
+import java.util.Objects;
+
+/**
+ * Executes one or several consecutive workflow auto transitions, i.e. a cascade
+ * of auto transitions, for an {@link AGWorkflowItem}.
+ *
+ * @author Alexander Schmidt
+ */
+public class WorkflowAutoTransitionCascadedExecutor {
+
+	private static final int DEFAULT_CASCADE_LENGTH_LIMIT = 100;
+
+	private final AGWorkflowItem item;
+	private final WorkflowAutoTransitionExecutor executor;
+	private int cascadeLengthLimit = DEFAULT_CASCADE_LENGTH_LIMIT;
+
+	/**
+	 * Creates a new instance.
+	 *
+	 * @param item
+	 *            the {@link AGWorkflowItem} to execute auto transitions for
+	 *            (never <i>null</i>)
+	 */
+	public WorkflowAutoTransitionCascadedExecutor(AGWorkflowItem item) {
+
+		this.item = Objects.requireNonNull(item);
+		this.executor = new WorkflowAutoTransitionExecutor(item);
+	}
+
+	/**
+	 * Defines the maximum length of an auto transition cascade.
+	 *
+	 * @param cascadeLengthLimit
+	 * @return this
+	 */
+	public WorkflowAutoTransitionCascadedExecutor setCascadeLengthLimit(int cascadeLengthLimit) {
```

Comment:
possible alternative naming `setRecursionLimit`

Reply:
Everything so far is dubbed "cascade". I think the the introduction of a synonym would not be beneficial.